stdpopsim/slim_engine.py

```diff
@@ -648,6 +697,92 @@ def slim_array_string(iterable, indent, width=80):
     )
 
 
+def _simulate_mutation_type_in_slim(mt):
+    """
+    Returns true if this `stdpopsim.MutationType` is to be simulated by SLiM.
+    """
+    return not mt.is_neutral
+
+
+def _enum_dfe_and_intervals(contig):
+    """
+    Returns an iterator which, for every `DFE` in the `contig`, returns a tuple
+    containing a numeric id for said DFE, the DFE object and the list of
+    intervals associated with the DFE.
+    """
+    for i, (d, ints) in enumerate(zip(contig.dfe_list, contig.interval_list)):
+        yield i, d, ints
+
+
+def _dfe_to_mtypes_dict(contig):
+    """
+    Assigns a mutation type id to each of the mutation types contained in this
+    Contig. This function will return a dictionary with `len(contig.dfe_list)`
+    elements with the DFE ids as keys. For each DFE id, the dictionary holds a
```

Comment:
keys are not dfe ids, but the position of the dfe in the list.

Reply:
should be list, not a dict?

Reply:
I think to be consistent with the other steps that use `_enum_dfe_and_intervals` we should leave this as a dictionary. In practice it does the same thing a list would do, but in the future if we decide to change the way we index DFEs it would be easier to change.

Reply:
Ok, I guess? It seems confusing to have a dict indexed by integers. At least, the docstring needs to be fixed.